aptos-move/framework/aptos-framework/sources/AptosGovernance.move

```diff
@@ -248,6 +263,13 @@ module AptosFramework::AptosGovernance {
         );
     }
 
+    /// Return a signer for making changes to 0x1 as part of on-chain governance proposal process.
+    /// Accept a GovernanceProposal object instead of a reference so it can't be used more than once.
+    public fun get_framework_signer(_proposal: GovernanceProposal): signer acquires GovernanceResponsbility {
```

Comment:
How can you prevent that this function is called by unauthorized code?

Reply:
PS. Should perhaps use friends here?

Reply:
It needs GovernanceProposal so as we discussed earlier today, this cannot be called outside of the governance process. This cannot be friend because of the resolution script needs to call it and use the signer to make required changes.

Reply:
Right indeed -- the proposal really acts as a capability, which happen to be only acquirable inside of the proposal script. Lets perhaps later look at renaming this once we do the capability cleanup.